frontends/election-manager/src/utils/election.ts

```diff
@@ -174,6 +176,37 @@ export function getContestsForPrecinct(
   });
 }
 
+export function generateWriteIn(position: number): WriteInCandidate {
+  return {
+    id: 'write-in',
+    isWriteIn: true,
+    name: 'WRITE-IN',
+    writeInPosition: position,
+  };
+}
+
+export function numBallotPositions(contest: AnyContest): number {
+  if (contest.type === 'candidate') {
+    return (
+      contest.candidates.length + (contest.allowWriteIns ? contest.seats : 0)
+    );
+  }
+  return 2;
+}
+
+export function candidateAtPosition(
+  contest: CandidateContest,
+  position: number
+): Candidate {
+  assert(position < numBallotPositions(contest)); // safety check
+  if (position < contest.candidates.length) {
+    return contest.candidates[position];
```

Comment:
I don't think this will work with [`CandidateContest#rotation`](https://github.com/votingworks/vxsuite/blob/7380af20d365f142f5ac32da536430faf135577f/libs/types/src/election.ts#L199).

Reply:
No, I'm fine with this as long as rotation does work. Probably should have a test to that effect.

Reply:
Oh, I forgot it's only for BMDs 🤦 Carry on.